flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/ZooKeeperLeaderRetrievalService.java

```diff
@@ -100,6 +100,12 @@ public void start(LeaderRetrievalListener listener) throws Exception {
 		LOG.info("Starting ZooKeeperLeaderRetrievalService {}.", retrievalPath);
 
 		synchronized (lock) {
+			// A double check for the race condition that multi invocation on this method
+			// could pass the first checkState() and wait on the lock, then they can all reset
+			// the leaderListener.
+			Preconditions.checkState(leaderListener == null, "ZooKeeperLeaderRetrievalService can " +
```

Comment:
This change is unrelated?

Reply:
Ah, this is a hotfix...I would like to pull it out into another new PR if you object it here...

Reply:
@zentol what do you think?